### One Tailed (Right Tailed) - One-Sample Z-Test

**1 - What is it?**

A. `Right One-Tailed One-Sample Z-Test` check the means of sample is significantly higher from the population mean (when population standard devidation is known and sample size is large, usually n>=30).

`Think of i like` is my small group's of average weight `really higher(significantly)` from the entire country's average weight, or it just random chance.

**2- Real-Life Example : Right Tailed One-Sample Z-Test**

`Scenario:` Your marketing teams says: `Out new landing page increase average session time beyond 5 minutes`

So we wanted to test if its really greater than 5 minutes, not just difference - So it's one-tailed test (specifically, right-tailed)

**3 - Hypotheses**

- H0 (Null Hypothesis) : Population mean (meau) = 5 (no improvement).
- H1 (Alternate Hypothesis) : Sample mean > Population mean (mean time improved)

This is a `right-tailed` test because we're checking the `greater than`.

**4 - Formula**

Z = (Sample mean - Population mean) / (Population std/Sqrt(Sample Size))

Z = (x bar - μ) / σ/sqrt(n)

**Parameters:**
- Population mean (μ) : 5 min
- Population std (σ) : 1 min
- Sample mean (x bar) : 5.5 min
- Sample size (n) : 50
- Significance level (alpha) : 0.05
- confidence Interval (CI) : 0.95

Here `population std is know and sample size>=30` so we can do z-test, specifically *right one-tailed one-sample z-test*.

**5 - Python Implementation (Step-by_Step)**

In [3]:
import numpy as np
from scipy import stats

# -------------------------------------------------------
# Step 1: Inputs - These are the numbers we already know
# -------------------------------------------------------

population_mean = 5     # Our usual average session time
population_std  = 1     # How much people's times usually vary
sample_mean     = 5.5   # Average time after the new landing page
sample_size     = 50    # Number of users in our test
alpha           = 0.05  # 5% chance we allow for being wrong

# ---------------------------------------------------------
# Step 2: Standard Error (SE)
# SE tell us how much the sample mean is expected to wiggle
# Just because of random variation.
# ----------------------------------------------------------

standard_error = population_std / np.sqrt(sample_size)
print(f"Standard Error (SE): {standard_error:.4f}")

# -----------------------------------------------------------
# Step 3: Z-Statistic
# This tell us how far our new mean (5.5) is from the old mean (5)
# Measured in units of SE.
# Bigger z = stronger evidence that the increase is real.
# -----------------------------------------------------------

z_stat = (sample_mean - population_mean)/standard_error
print(f"Z Statistic: {z_stat:.4f}")

# ------------------------------------------------------------
# Step 4: P-value (right-tailed)
# This answers: "If there was NO real improvement,
# What's the chance of seeing a result this high purely by luchk?"
# Small p-value (<alpha) = unlikely to be luck  = improvement is real.
# --------------------------------------------------------------

p_value = 1-stats.norm.cdf(z_stat)
print(f"P-Value: {p_value:.4f}")

# --------------------------------------------------------------
# Step 5: Critical Value
# This is the cutoff point for the signficance.
# For a = 0.05 right-tailed -> critical z = 1.645
# If z_stat is above, it's considered statistically high enough.
# ---------------------------------------------------------------

critical_value = stats.norm.ppf(1 - alpha)
print(f"Critical Value: {critical_value:.4f}")

# ---------------------------------------------------------------
# Step 6: Decision Based on P-Value
# ---------------------------------------------------------------

print("\n-------- Decision Based on P-Value -----------")
if p_value < alpha:
    print("Reject H0: The new landing page significantly increased session")
else:
    print("Fail to reject H0: No strong evidence of improvement.")

# -----------------------------------------------------------------
# Step 7: Decision Based on Z-Value
# -----------------------------------------------------------------

print("\n--------- Decision Based on Critical Value ---------")
if z_stat >  critical_value:
    print("Reject H0: The new landing page significantly increased  session")
else:
    print("Fail to Reject H0: No Strong evidence of improvement.")

Standard Error (SE): 0.1414
Z Statistic: 3.5355
P-Value: 0.0002
Critical Value: 1.6449

-------- Decision Based on P-Value -----------
Reject H0: The new landing page significantly increased session

--------- Decision Based on Critical Value ---------
Reject H0: The new landing page significantly increased  session


**6 - Business Interpretation**

- Z = 3.53 -> Sample mean is **3.5 standard error above** the population mean.
- p = 0.0002 -> There is only **0.02% chance** that this result happened by random luck.
- Therefore, the **new landing page truly increased** session time, the difference is **statistically significant**, not random.